paper/src/main/java/vg/civcraft/mc/namelayer/database/GroupManagerDao.java

```diff
@@ -493,8 +457,61 @@ public Boolean call() {
 				"DELETE FROM permissionByGroup "
 						+ "WHERE role='" + PlayerType.NOT_BLACKLISTED +"' "
 						+ "AND perm_id=(SELECT perm_id FROM permissionIdMapping WHERE name='BASTION_PLACE');");
+
+		db.registerMigration(15, false,
+				"ALTER TABLE faction ADD COLUMN id INT NOT NULL FIRST",
+				"UPDATE faction, faction_id SET faction.id = faction_id.group_id WHERE faction.group_name = faction_id.group_name",
+				"ALTER TABLE faction DROP PRIMARY KEY",
+				"ALTER TABLE faction ADD CONSTRAINT UNIQUE KEY (group_name)",
+				"ALTER TABLE faction ADD PRIMARY KEY (id)",
+				"ALTER TABLE faction MODIFY id INT NOT NULL AUTO_INCREMENT",
+				"DROP TABLE faction_id",
+				"DROP TABLE subgroup", // don't use old values from subgroup as it has been disabled anyway
+				"ALTER TABLE faction ADD COLUMN (parent_id INT)",
+				// TODO: ON DELETE SET NULL or ON DELETE CASCADE? If the parent group is deleted, should the child group be unlinked or deleted too?
```

Comment:
my vote would be to orphan children, less destructive if parent delete was unintended or part of a reorganization